src/main/java/com/github/martapietka/ppm2pbm/ByteReader.java

```diff
@@ -7,19 +7,18 @@ public class ByteReader {
 
     public static int convertBytesToInt(InputStream inputStream) throws IOException {
 
-        byte[] array = inputStream.readAllBytes();
-
         StringBuilder sb = new StringBuilder();
 
-        for (int a=0; a < array.length; a++) {
-            if (array[a] <= 0x1F) {
+        int number;
+        while ((number = inputStream.read()) != -1) {
+            if (number <= 0x1F) {
                 break;
             }
-            int digit = array[a] - 48;
-            sb.append(digit);
+            number -= 48;
```

Comment:
Chyba bym został ze zmienną pomocniczą `digit` i nie zmieniał wartości zmiennej `number` (lub innej, zależnie jak ją nazwiesz jeśli zastosujesz sugestię powyżej). Działanie to samo, ale dużo wyższa czytelność.

Także ogólnie unikałbym zapisu `-=` - jest krótki i to w zasadzie jego jedyna zaleta 😉 powszechnie raczej uważa się go za zaburzającego czytelność.